### ISSUE TITLE
Resque worker shuts down when a terminal or ssh connection exits

### ISSUE THREAD

**Issue by nisanth074** [nevans/resque-pool#115]
With resque-pool, a resque worker started from a terminal shuts down when the terminal exits
```
nohup bundle exec rake resque:work &
```

When the terminal exits, the worker's ppid becomes 1. The worker thinks its pool master has died (even though it isn't part of any pool) and shuts down.
```ruby
# https://github.com/nevans/resque-pool/blob/v0.3/lib/resque/pool/pooled_worker.rb
module PooledWorker
  def shutdown_with_pool
    shutdown_without_pool || Process.ppid == 1
  end  
# ...
```
We need to query the running status of the pool master only when the worker is part of a pool.

A possible fix (borrowed from previous commits) is to save the process pid when Resque::Worker is initialized and use it later to determine if the worker is part of a pool.
```ruby
def initialize(*args)
  @__original_pid = Process.pid
end

alias_method :__shutdown?, :shutdown?
def shutdown?
  return true if __shutdown?
  pooled? && pool_master_has_gone_away?
end

private

def pooled?
  Process.pid != @__original_pid
end

def pool_master_has_gone_away?
  Process.ppid == 1
end
```
I've fixed this in a fork (https://github.com/SupportBee/resque-pool/commit/88d2537f49b247611ec7545b8f4d201eb2b4ca57) and would be happy to send in a pull request. Thanks.

**Comment by nevans** [comment]
Sorry for taking so long to respond to this.

1. It seems strange that `resque/pool/pooled_worker` is being loaded when the worker isn't actually being managed by Resque::Pool.  Have you traced why that is happening?
2. However I've recently been trying to get my environment running under docker, which runs the container command with PID=1.  And *that* certainly messes up this logic too.  I'll use some of your code (`pooled?` in particular) so hopefully it should solve your problem as well.

**Comment by nisanth074** [comment]
@nevans
> 1. It seems strange that resque/pool/pooled_worker is being loaded when the worker isn't actually being managed by Resque::Pool. Have you traced why that is happening?

I'm managing gems with Bundler, and I hadn't passed the `:require => false` option when adding resque-pool to my Gemfile
```ruby
# My Gemfile had
gem 'resque-pool'
# instead of
gem 'resque-pool', :require => false
```
With `:require => false`, `resque/pool/pooled_worker` isn't loaded and this issue won't crop up

> 2. However I've recently been trying to get my environment running under docker, which runs the container command with PID=1. And that certainly messes up this logic too. I'll use some of your code (pooled? in particular) so hopefully it should solve your problem as well.

Thanks, that would be nice

**Comment by nevans** [comment]
I don't *think* that this is fixed by #118, now on master and in 0.4.0-rc3 (which will be released as 0.4.0 next week if it all works out).  I didn't change how things were required, so we will still monkeypatch `Resque::Worker` as soon as you require `resque/pool`.  Can you re-write your patch based on current master?  Thanks!

**Comment by nevans** [comment]
I think I fixed this in 1edc803 but I haven't fully tested your scenario yet (I just tested that it didn't break everything else).  Can you try out `master` and let me know if it works for you?

**Comment by nisanth074** [comment]
@nevans https://github.com/nevans/resque-pool/commit/1edc80327d71331d29530cdeb80407ce1178dafe fixes the issue :+1:
